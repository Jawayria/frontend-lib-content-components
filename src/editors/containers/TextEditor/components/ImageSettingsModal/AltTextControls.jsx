import React from 'react';
import PropTypes from 'prop-types';

import { Form } from '@edx/paragon';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import * as hooks from './hooks';
import messages from './messages';

/**
 * Wrapper for alt-text input and isDecorative checkbox control
 * @param {bool} isDecorative - is the image decorative?
 * @param {func} setIsDecorative - handle isDecorative change event
 * @param {func} setValue - update alt-text value
 * @param {string} value - current alt-text value
 */
export const AltTextControls = ({
  isDecorative,
  setIsDecorative,
  setValue,
  value,
  // inject
  intl,
}) => (
  <Form.Group className="mt-4.5">
    <Form.Label as="h4">
      <FormattedMessage {...messages.accessibilityLabel} />
    </Form.Label>
    <Form.Control
      className="mt-4.5"
      type="input"
      value={value}
      disabled={isDecorative}
      onChange={hooks.onInputChange(setValue)}
      floatingLabel={intl.formatMessage(messages.altTextFloatingLabel)}
    />
    <Form.Checkbox
      className="mt-4.5 decorative-control-label"
      checked={isDecorative}
      onChange={hooks.onCheckboxChange(setIsDecorative)}
    >
      <Form.Label>
        <FormattedMessage {...messages.decorativeCheckboxLabel} />
      </Form.Label>
    </Form.Checkbox>
  </Form.Group>
);
AltTextControls.propTypes = {
  isDecorative: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
  setValue: PropTypes.func.isRequired,
  setIsDecorative: PropTypes.func.isRequired,
  // inject
  intl: intlShape.isRequired,
};

export default injectIntl(AltTextControls);
