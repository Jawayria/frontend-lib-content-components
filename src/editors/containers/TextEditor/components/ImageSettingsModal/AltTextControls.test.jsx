import React from 'react';
import { shallow } from 'enzyme';

import { formatMessage } from '../../../../../testUtils';
import { AltTextControls } from './AltTextControls';

jest.mock('./hooks', () => ({
  onInputChange: (handler) => ({ 'hooks.onInputChange': handler }),
  onCheckboxChange: (handler) => ({ 'hooks.onCheckboxChange': handler }),
}));

describe('AltTextControls', () => {
  const props = {
    isDecorative: true,
    value: 'props.value',
    // inject
    intl: { formatMessage },
  };
  beforeEach(() => {
    props.setValue = jest.fn().mockName('props.setValue');
    props.setIsDecorative = jest.fn().mockName('props.setIsDecorative');
  });
  describe('render', () => {
    test('snapshot: isDecorative=true', () => {
      expect(shallow(<AltTextControls {...props} />)).toMatchSnapshot();
    });
  });
});
