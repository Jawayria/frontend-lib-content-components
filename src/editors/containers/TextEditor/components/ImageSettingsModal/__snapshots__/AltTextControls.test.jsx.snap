// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltTextControls render snapshot: isDecorative=true 1`] = `
<Form.Group
  className="mt-4.5"
>
  <Form.Label
    as="h4"
  >
    <FormattedMessage
      defaultMessage="Accessibility"
      description="Label title for accessibility section."
      id="authoring.texteditor.imagesettingsmodal.accessibilityLabel"
    />
  </Form.Label>
  <Form.Control
    className="mt-4.5"
    disabled={true}
    floatingLabel="Alt Text"
    onChange={
      Object {
        "hooks.onInputChange": [MockFunction props.setValue],
      }
    }
    type="input"
    value="props.value"
  />
  <Form.Checkbox
    checked={true}
    className="mt-4.5 decorative-control-label"
    onChange={
      Object {
        "hooks.onCheckboxChange": [MockFunction props.setIsDecorative],
      }
    }
  >
    <Form.Label>
      <FormattedMessage
        defaultMessage="This image is decorative (no alt text required)."
        description="Checkbox label for whether or not an image is decorative."
        id="authoring.texteditor.imagesettingsmodal.decorativeCheckboxLabel"
      />
    </Form.Label>
  </Form.Checkbox>
</Form.Group>
`;
